spec/formula_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'spec_helper.rb'
+
+describe Statsample::GLM::Formula do
+  context '#parse_formula' do
+    context 'no interaction' do
+      let(:formula) { described_class.new 'y ~ a+b' }
+      subject { formula.parse_formula }
+      
+      it { is_expected.to be_a Array }
+      its(:first) { is_expected.to eq Statsample::GLM::Token }
+      its(:to_basic_formula) { is_expected.to eq 'y ~ 1+a(-)+b(-)' }
```

Comment:
@zverok We might need to relate the parsed formula terms with input terms, so we need to store the input terms. I have included an option `form` in `#parse_formula` so when its `form=:string` it works like `#to_basic_formula`.